db/keyval/filedb/filesystem/filesystem.go

```diff
@@ -0,0 +1,180 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package filesystem
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/pkg/errors"
+)
+
+// API defines filesystem-related method with emphasis on the fileDB needs
+type API interface {
+	// CreateFile creates a new file. Returns error if provided parameter is directory.
+	CreateFile(file string) error
+	// ReadFile returns an content of given file
+	ReadFile(file string) ([]byte, error)
+	// WriteFile writes data to file
+	WriteFile(file string, data []byte) error
+	// FileExists returns true if given path was found
+	FileExists(file string) bool
+	// GetFiles takes a list of file system paths an returns a list of individual file paths
+	GetFileNames(paths []string) ([]string, error)
+	// Watch given paths, call 'onEvent' function on every event or 'onClose' if watcher is closed
+	Watch(paths []string, onEvent func(event fsnotify.Event), onClose func()) error
+	// Close allows to terminate watcher from outside which calls 'onClose'
+	Close() error
+}
+
+// FsHandler is helper struct to manipulate with filesystem API
+type FsHandler struct {
```

Comment:
The `Fs` prefix sounds like unnecessary stutter when used: `filesystem.FsHandler`. I think `filesystem.Handler` would suffice here.